docs/proposals/proposal-003-report.md

```diff
@@ -0,0 +1,275 @@
+- Integration with cncf dev-stat on Grafana
+
+### Linked Docs
+
+<!--
+Provide links to previous discussions, Slack threads, motivation issues or any other document
+with context. It is really helpful to provide a "source of truth" for the work
+so that people aren't searching all over the place for lost context.
+-->
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is:
+outlining your solution to the problem described in the Motivation section.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this proposal is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+**CNCF project maintainer selects project-related metrics** 
+
+If the project produces significant metrics that need to be monitored along with the sustainable metrics, I would like to have them reported.
+
+**Green reviews maintainer adds, modifies or removes sustainability metrics**
+
+As a Green Review maintainer, I would like to change the sustainability metrics over time. 
+
+**CNCF project maintainer is able to check the metrics for their project**
+
+After the pipeline produces the metrics, I would like to see the result of it in an accessible way.
+
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate?
+Think broadly.  For example, consider how this will impact or be impacted
+by scaling to support more CNCF Projects.
+
+How will this affect the benchmark tests, CNCF Project Maintainers, pipeline maintainers, etc?
+-->
+
+As with every design document there are some challanges:
+ 
+- Consistency vs Flexibility: if we change the sustainability metrics overtime it will be difficult to compare different metrics from different green reviews release. However we would rather be flexible in this first phase and possibly change what we store and how if this leads to more correct results. 
+- The three sub-steps of the proposal: Collect, Store and Share are co-dependent. How to collect the data depends on how to store it, and how to store the data depends on how to show it. Since we are still in early phases of the working group, an agile approach will be proposed: a first lean solution will be deployed and, most likely, improved in the future.  
+
+## Design Details
+
+This section will have the following subsections:
+
+- Metrics: what metrics to collect?
+- Collect: how to collect the metrics?
+- Store: how to store them?
+- Share: how to share the metrics?
+
+### Metrics
+
+As already mentioned we will have two sets of metrics:
```

Comment:
> So we don't expect high volumes of data (last famous words ;))

@locomundo Good list and yes I'm not expecting high data volumes since its per pipeline run. At least not right now! 😅 

> (To take the Falco example, it is mentioned in this proposal that they also need rate(container_cpu_usage_seconds_total[5m]), container_memory_rss and container_memory_working_set_bytes)
(are these always extracted from Kepler? I suppose, for now they are because it's the only measurement tool we use).

The SRE type metrics Falco have requested are cAdvisor metrics being scraped from the kubelet.
https://kubernetes.io/docs/reference/instrumentation/metrics/

For me this is an argument for querying Prometheus from the pipeline rather than scraping Kepler directly. As potentially we could then support any metric available in Prometheus.

Reply:
I think we should by default store the data about where we are performing the benchmarking tests and generally on what machine type, also some additional data and I think SCI score gets stored as well 🤔 

because we are treating these metrics are artifacts we should have these as well for information sake as well as if any project wants to verify our platform of benchmarking

Reply:
@dipankardas011 Yes, we should record metadata about the pipeline run. 

Good point to mention SCI as so far its not covered in the proposal. If we continue to calculate an SCI score I think we should include the other inputs such as embodied carbon in the stored data.

Storing the underlying metrics adds some complexity but I can see the benefits for verifying our results. Short term the metrics will be available in Prometheus until the retention period.